kong/plugins/oauth2/access.lua

```diff
@@ -95,7 +95,12 @@ local function retrieve_parameters()
   ngx.req.read_body()
 
   -- OAuth2 parameters could be in both the querystring or body
-  return utils.table_merge(ngx.req.get_uri_args(), public_utils.get_body_args())
+  local uri_args = ngx.req.get_uri_args()
+  local body_args = {}
+  if ngx.req.get_method() ~= "GET" then
```

Comment:
Why create the body args table outside this condition? It's a needless table creation and gc on GET requests. We should rather create this temp table and merge it with uri args only when the req method is appropriate

Reply:
thanks, I'll fix it